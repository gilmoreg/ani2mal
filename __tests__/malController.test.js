/* eslint-disable no-unused-vars */
/* globals describe, it, beforeEach, afterEach, expect */
const chai = require('chai');
const chaiHttp = require('chai-http');
const fetchMock = require('fetch-mock');
const malController = require('../controllers/mal');
const fakes = require('../helpers/fakes');

const { app, runServer, closeServer } = require('../server');

chai.use(chaiHttp);

describe('malController', () => {
  beforeEach(() => {
    runServer(5000);
  });
  afterEach(() => {
    fetchMock.restore();
    closeServer();
  });

  it('Should update an anime if already in the list', (done) => {
    fetchMock.mock('*', new Error('unmatched fetch'));
    fetchMock.mock(/.+animelist\/add.+/g, fakes.malAddSuccess1);
    fetchMock.mock(/.+animelist\/update.+/g, fakes.malUpdateSuccess);
    chai.request(app)
      .post('/mal/add')
      .send({
        auth: 'auth',
        anilist: fakes.aniList,
      })
      .then((res) => {
        expect(res.body).toMatchSnapshot();
        done();
      });
  });
});

