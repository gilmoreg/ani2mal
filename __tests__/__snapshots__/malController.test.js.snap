// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`malController Should update a manga if it is already in the list 1`] = `"Updated"`;

exports[`malController Should update an anime if already in the list 1`] = `"Updated"`;
